lib/services/mappingService.ts

```diff
@@ -1,33 +1,25 @@
-import {CDEStatus, CDEType, DatasetCDEMapping, InputMapping} from "../models.ts";
+import {CDEMapping, CDEStatus, CDEType, InputMappingRow, MappingRow} from "../models.ts";
 
-export const validateInputMappings = (inputMappings: InputMapping[][]) => {
-    return inputMappings.every(innerArray =>
-        innerArray.every(mapping =>
-            'variableName' in mapping &&
-            'abbreviation' in mapping &&
-            'interlexID' in mapping
-        )
-    );
+export const areMappingsValid = (mappings: InputMappingRow[]) => {
+    return mappings.length > 0 && mappings.every(mapping =>
+        'variableName' in mapping &&
+        'abbreviation' in mapping &&
+        'interlexID' in mapping
+    )
 };
 
+export const getMappingsDict = (inputMappingRows: InputMappingRow[]): CDEMapping => {
+    const cdeMapping: CDEMapping = {};
```

Comment:
question: @Aiga115 imho the data should be in the context. So I propose to use the init to call the context to do this kind of stuff

Reply:
This was actually my doing @zsinnema. I think we can discuss it further in our meeting today but the general idea was to not have the initial mapping provided by the customer as part of the context but rather a superset of it, where the some non-explicit information like the `status` and the `cdeType` are made explicit. The goal of the `getMappingsDict` method is to do such conversion.

Reply:
From your other comments I realized that the input mappings are part of the context right now, I would say that's the bit that needs to change but let me know if you agree @zsinnema

Reply:
@afonsobspinto imo the status and other attribs ARE part of the application state of the dataset mapping (which is initially initted by the init "create context" from the initial mapping). F.e. the context dataset mapping is needed for (checking/validation) the "export" back to the website. Or do I miss what you are saying?